Missing note for duplicate global actor annotations

### Description

There should be a note indicating where the other annotations are in source.

### Reproduction

```swift
@MainActor // func uhoh-commented-out() {
//   print("oops")
// }
@MainActor func myFunction() {}
```

```shell
➜  Desktop (main 6/9/25) swiftc -c test.swift
test.swift:4:17: error: declaration can not have multiple global actor attributes ('MainActor' and 'MainActor')
2 | //   print("oops")
3 | // }
4 | @MainActor func myFunction() {}
  |                 `- error: declaration can not have multiple global actor attributes ('MainActor' and 'MainActor')
5 |
6 |
```

### Expected behavior

a note referring to source location 1:1.

### Environment

```shell
$ swiftc --version
Apple Swift version 6.2-dev (LLVM 3c4a54b02dd62b7, Swift a40a7be6945a895)
Target: arm64-apple-macosx16.0
Build config: +assertions
```

### Additional information

_No response_

--- a/include/swift/AST/DiagnosticsSema.def
+++ b/include/swift/AST/DiagnosticsSema.def
@@ -6228,6 +6228,8 @@ ERROR(no_concurrency_module,none,
 ERROR(multiple_global_actors,none,
       "declaration can not have multiple global actor attributes (%0 and %1)",
       (Identifier, Identifier))
+NOTE(multiple_global_actors_note,none,
+      "%0 attribute previously declared here", (Identifier))
 ERROR(global_actor_disallowed,none,
       "%kindonly0 cannot have a global actor", (const Decl *))
 ERROR(global_actor_on_actor_class,none,

--- a/include/swift/AST/DiagnosticsSema.def
+++ b/include/swift/AST/DiagnosticsSema.def
@@ -6229,7 +6229,7 @@ ERROR(multiple_global_actors,none,
       "declaration can not have multiple global actor attributes (%0 and %1)",
       (Identifier, Identifier))
 NOTE(multiple_global_actors_note,none,
-      "%0 attribute previously declared here", (Identifier))
+      "%kindonly0 attribute previously declared here", (const Decl *))
 ERROR(global_actor_disallowed,none,
       "%kindonly0 cannot have a global actor", (const Decl *))
 ERROR(global_actor_on_actor_class,none,

--- a/include/swift/AST/DiagnosticsSema.def
+++ b/include/swift/AST/DiagnosticsSema.def
@@ -6229,7 +6229,7 @@ ERROR(multiple_global_actors,none,
       "declaration can not have multiple global actor attributes (%0 and %1)",
       (Identifier, Identifier))
 NOTE(multiple_global_actors_note,none,
-      "%kindonly0 attribute previously declared here", (const Decl *))
+      "%0 attribute previously declared here", (const Decl *))
 ERROR(global_actor_disallowed,none,
       "%kindonly0 cannot have a global actor", (const Decl *))
 ERROR(global_actor_on_actor_class,none,

--- a/lib/Sema/TypeCheckConcurrency.cpp
+++ b/lib/Sema/TypeCheckConcurrency.cpp
@@ -340,6 +340,10 @@ swift::checkGlobalActorAttributes(SourceLoc loc, ArrayRef<CustomAttr *> attrs) {
       ctx.Diags.diagnose(
           loc, diag::multiple_global_actors, globalActorNominal->getName(),
           nominal->getName());
+
+      ctx.Diags.diagnose(attr->getLocation(), diag::multiple_global_actors_note,
+                         nominal->getName());
+
       continue;
     }
 

--- a/lib/Sema/TypeCheckConcurrency.cpp
+++ b/lib/Sema/TypeCheckConcurrency.cpp
@@ -342,7 +342,7 @@ swift::checkGlobalActorAttributes(SourceLoc loc, ArrayRef<CustomAttr *> attrs) {
           nominal->getName());
 
       ctx.Diags.diagnose(attr->getLocation(), diag::multiple_global_actors_note,
-                         nominal->getName());
+                         nominal);
 
       continue;
     }

--- a/test/attr/global_actor.swift
+++ b/test/attr/global_actor.swift
@@ -88,12 +88,12 @@ class SomeClass {
 @GA1 actor ActorInTooManyPlaces { } // expected-error{{actor 'ActorInTooManyPlaces' cannot have a global actor}}
 
 @GA1 @OtherGlobalActor func twoGlobalActors() { } // expected-error{{declaration can not have multiple global actor attributes ('OtherGlobalActor' and 'GA1')}}
-// expected-note@-1{{struct attribute previously declared here}}
+// expected-note@-1{{'GA1' attribute previously declared here}}
 
 struct Container {
   // FIXME: Diagnostic could be improved to show the generic arguments.
 @GenericGlobalActor<Int> @GenericGlobalActor<String> func twoGenericGlobalActors() { } // expected-error{{declaration can not have multiple global actor attributes ('GenericGlobalActor' and 'GenericGlobalActor')}}
-// expected-note@-1{{generic struct attribute previously declared here}}
+// expected-note@-1{{'GenericGlobalActor' attribute previously declared here}}
 }
 
 // -----------------------------------------------------------------------

--- a/test/attr/global_actor.swift
+++ b/test/attr/global_actor.swift
@@ -88,12 +88,12 @@ class SomeClass {
 @GA1 actor ActorInTooManyPlaces { } // expected-error{{actor 'ActorInTooManyPlaces' cannot have a global actor}}
 
 @GA1 @OtherGlobalActor func twoGlobalActors() { } // expected-error{{declaration can not have multiple global actor attributes ('OtherGlobalActor' and 'GA1')}}
-// expected-note@-1{{'GA1' attribute previously declared here}}
+// expected-note@-1{{struct attribute previously declared here}}
 
 struct Container {
   // FIXME: Diagnostic could be improved to show the generic arguments.
 @GenericGlobalActor<Int> @GenericGlobalActor<String> func twoGenericGlobalActors() { } // expected-error{{declaration can not have multiple global actor attributes ('GenericGlobalActor' and 'GenericGlobalActor')}}
-// expected-note@-1{{'GenericGlobalActor' attribute previously declared here}}
+// expected-note@-1{{generic struct attribute previously declared here}}
 }
 
 // -----------------------------------------------------------------------

--- a/test/attr/global_actor.swift
+++ b/test/attr/global_actor.swift
@@ -88,10 +88,12 @@ class SomeClass {
 @GA1 actor ActorInTooManyPlaces { } // expected-error{{actor 'ActorInTooManyPlaces' cannot have a global actor}}
 
 @GA1 @OtherGlobalActor func twoGlobalActors() { } // expected-error{{declaration can not have multiple global actor attributes ('OtherGlobalActor' and 'GA1')}}
+// expected-note@-1{{'GA1' attribute previously declared here}}
 
 struct Container {
   // FIXME: Diagnostic could be improved to show the generic arguments.
 @GenericGlobalActor<Int> @GenericGlobalActor<String> func twoGenericGlobalActors() { } // expected-error{{declaration can not have multiple global actor attributes ('GenericGlobalActor' and 'GenericGlobalActor')}}
+// expected-note@-1{{'GenericGlobalActor' attribute previously declared here}}
 }
 
 // -----------------------------------------------------------------------

